src/pages/library/datasetBuilder/DatasetBuilder.ts

```diff
@@ -409,10 +409,35 @@ export const DatasetBuilderContents = ({ onStateChange }: { onStateChange: OnSta
       }),
       h(ConceptSetSelector, {
         selectedConceptSets,
-        onChange: (conceptSets) => setSelectedConceptSets(conceptSets),
+        onChange: async (conceptSets) => {
+          setSelectedConceptSets(conceptSets);
+          const uniqueFeatureValueGroups = _.flow(
+            _.flatMap((headerAndValues: HeaderAndValues<ConceptSet>) => headerAndValues.values),
+            _.map((conceptSet: ConceptSet) => conceptSet.featureValueGroupName),
+            _.uniq
+          )(conceptSets);
+          setValues(
+            _.flow(
+              _.filter((featureValueGroup: FeatureValueGroup) =>
+                _.includes(featureValueGroup.name, uniqueFeatureValueGroups)
+              ),
+              _.sortBy('name'),
+              _.map((featureValueGroup: FeatureValueGroup) => ({
+                header: featureValueGroup.name,
+                values: featureValueGroup.values,
+              }))
+            )(dataset.featureValueGroups)
+          );
+          setSelectedValues(
+            _.filter(
+              (selectedValueGroup) => _.includes(selectedValueGroup.header, uniqueFeatureValueGroups),
+              selectedValues
+            )
+          );
+        },
         onStateChange,
       }),
-      h(ValuesSelector, { selectedValues, values: [], onChange: (values) => setSelectedValues(values) }),
+      h(ValuesSelector, { selectedValues, values, onChange: (values) => setSelectedValues(values) }),
```

Comment:
I think this is the same as
```suggestion
      h(ValuesSelector, { selectedValues, values, onChange: setSelectedValues }),
```